### PR TITLE
Non-OS implementation of sleep_for() routine along with stm32f4xx counterpart 

### LINKED ISSUE #70: stm32f4xx:  this_thread namespace for non-os environments is missing
[These functions](https://github.com/forGGe/theCore/blob/master/lib/thread/freertos/export/os/utils.hpp) are implemented for FreeRTOS, but it missing [in the common thread library](https://github.com/forGGe/theCore/tree/master/lib/thread/common)

Such routines must be implemented.

### PR DESCRIPTION
**Pull request by forGGe**
Simple busy-wait loop is used. Closes #70

### CODE REVIEW

File: lib/thread/freertos/signal.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#include <os/signal.hpp>
+#include <ecl//thread/signal.hpp>
```
Comment (vadimol) [inline lib/thread/freertos/signal.cpp]: mistype

File: lib/thread/default/export/ecl/thread/utils.hpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
+//! \file
+//! \brief Default utilities.
+#ifndef LIB_THREAD_DEFAULT_UTILS_HPP_
+#define LIB_THREAD_DEFAULT_UTILS_HPP_
+
```
Comment (vadimol) [inline lib/thread/default/export/ecl/thread/utils.hpp]: just to clarify code style, should we use trailing underscore in define "guards?"

File: platform/stm32f4xx/export/platform/execution.hpp [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+//! \details Uses DWT register.
+//! See http://www.carminenoviello.com/2015/09/04/precisely-measure-microseconds-stm32/
+//! \param[in] ms Milliseconds to wait
+void spin_wait(uint32_t ms)
+{
+    auto start = DWT->CYCCNT;
+    auto to_wait = ms * (SystemCoreClock / 1000L);
+
+    // Handles wraparound as well.
+    do { } while (DWT->CYCCNT - start < to_wait);
```
Comment (vadimol) [inline platform/stm32f4xx/export/platform/execution.hpp]: possible overflow. CYCCNT wraps around to 0 on overflow according to ARM specification:  http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.ddi0337e/ch11s05s01.html ,
so we can handle it here.

**Comment by vadimol** [comment]
reviewed

File: lib/thread/default/export/ecl/thread/utils.hpp (hunk 2 above)
Comment (forGGe) [inline lib/thread/default/export/ecl/thread/utils.hpp]: Well, I can't say until we'll finish with #41 :)

File: platform/stm32f4xx/export/platform/execution.hpp (hunk 3 above)
Comment (forGGe) [inline platform/stm32f4xx/export/platform/execution.hpp]: Per discussion - overflow is handled here correctly

**Comment by forGGe** [comment]
Review done, merging...

**Pull request merged**